homeassistant/components/netatmo/__init__.py

```diff
@@ -1,286 +1,102 @@
-    except HTTPError:
-        _LOGGER.error("Unable to connect to Netatmo API")
-        return False
-
-    try:
-        home_data = pyatmo.HomeData(auth)
-    except pyatmo.NoDevice:
-        home_data = None
-        _LOGGER.debug("No climate device. Disable %s service", SERVICE_SETSCHEDULE)
+    config_flow.NetatmoFlowHandler.async_register_implementation(
+        hass,
+        config_entry_oauth2_flow.LocalOAuth2Implementation(
+            hass,
+            DOMAIN,
+            config[DOMAIN][CONF_CLIENT_ID],
+            config[DOMAIN][CONF_CLIENT_SECRET],
+            OAUTH2_AUTHORIZE,
+            OAUTH2_TOKEN,
+        ),
+    )
 
-    # Store config to be used during entry setup
-    hass.data[DATA_NETATMO_AUTH] = auth
+    return True
 
-    if config[DOMAIN][CONF_DISCOVERY]:
-        for component in "camera", "sensor", "binary_sensor", "climate":
-            discovery.load_platform(hass, component, DOMAIN, {}, config)
 
-    if config[DOMAIN][CONF_WEBHOOKS]:
-        webhook_id = hass.components.webhook.async_generate_id()
-        hass.data[DATA_WEBHOOK_URL] = hass.components.webhook.async_generate_url(
-            webhook_id
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Netatmo from a config entry."""
+    # Backwards compat
+    if "auth_implementation" not in entry.data:
+        hass.config_entries.async_update_entry(
+            entry, data={**entry.data, "auth_implementation": DOMAIN}
         )
-        hass.components.webhook.async_register(
-            DOMAIN, "Netatmo", webhook_id, handle_webhook
-        )
-        auth.addwebhook(hass.data[DATA_WEBHOOK_URL])
-        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, dropwebhook)
-
-    def _service_addwebhook(service):
-        """Service to (re)add webhooks during runtime."""
-        url = service.data.get(CONF_URL)
-        if url is None:
-            url = hass.data[DATA_WEBHOOK_URL]
-        _LOGGER.info("Adding webhook for URL: %s", url)
-        auth.addwebhook(url)
-
-    hass.services.register(
-        DOMAIN,
-        SERVICE_ADDWEBHOOK,
-        _service_addwebhook,
-        schema=SCHEMA_SERVICE_ADDWEBHOOK,
-    )
-
-    def _service_dropwebhook(service):
-        """Service to drop webhooks during runtime."""
-        _LOGGER.info("Dropping webhook")
-        auth.dropwebhook()
 
-    hass.services.register(
-        DOMAIN,
-        SERVICE_DROPWEBHOOK,
-        _service_dropwebhook,
-        schema=SCHEMA_SERVICE_DROPWEBHOOK,
+    implementation = await config_entry_oauth2_flow.async_get_config_entry_implementation(
+        hass, entry
     )
 
-    def _service_setschedule(service):
-        """Service to change current home schedule."""
-        schedule_name = service.data.get(ATTR_SCHEDULE)
-        home_data.switchHomeSchedule(schedule=schedule_name)
-        _LOGGER.info("Set home schedule to %s", schedule_name)
+    hass.data[DOMAIN][AUTH] = api.ConfigEntryNetatmoAuth(hass, entry, implementation)
```

Comment:
Store this as `hass.data[DOMAIN][entry.entry_id][AUTH]` so you can support multiple config entries.